MapboxVision/Services/VideoPlayer.swift

```diff
@@ -140,3 +141,28 @@ private extension CMSampleBuffer {
         return sampleBuffer
     }
 }
+
+extension VideoPlayer: VideoPlayerProtocol {
+    func addListener(_ listener: MBVVideoSourceObserver) {
+        nativeObservers.add(observer: listener)
+    }
+
+    func removeListener(_ listener: MBVVideoSourceObserver) {
+        nativeObservers.remove(observer: listener)
+    }
+
+    var progress: Float {
+        get {
+            let cmTime = player.currentTime()
+            return Float(Double(cmTime.value) / Double(cmTime.timescale))
+        }
+        set(progress) {
+            player.seek(to: CMTime.init(value: Int64(progress), timescale: 1))
```

Comment:
Timescale represents how precise we can "scroll" inside a second. Timescale 1 basically means losing the precision down to an integer value. A preferred value for movie clips is 600 to 90000 as documentation outlines.
```suggestion
            player.seek(to: CMTimeMakeWithSeconds(Double(newValue), preferredTimescale: 600))
```

Reply:
done